### DISCUSSION TITLE
Storybook - MDX - TypeScript - Props Autocompletion

### DISCUSSION THREAD

**Discussion by ghost** [storybookjs/storybook#16215] [General] [answered]
Hello guys,

I'm using MDX format to write my stories. I'm also using TypeScript.
Is there a way to benefit from TypeScript props autocompletion ?

It's not possible if I want to use the Template.bind() function (in order to get the Controls).

Thanks in advance

`
import Button from "../../src/Components/Button";

export const Template = (args) => (
  <Button {...args}  />
);

<Canvas>
  <Story
    name="Default"
    args={{
      textButton: "Beautiful Button",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

`

**Comment by kylegach** [comment, marked as answer]
Hi, @laplaceducss. I'd love to have that ability, too! Unfortunately, this is a limitation with MDX + Typescript. You can find more context here: https://github.com/microsoft/TypeScript/issues/36440.

**Reply by ghost** [reply]
Ah. So sad !!!

Thanks a lot for your answer, @kylegach :-)
